### ISSUE TITLE
Keyboard navigation in RadioGroup does not work as expected

### ISSUE THREAD

**Issue by khalibloo** [ant-design/ant-design#43306]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/3fx3xx)

### Steps to reproduce

1. Select item 2 in a radio group with 3 radio buttons
2. Focus on an element outside the radio group
3. Tab into the radio group

### What is expected?

The focus should be on Item 2 since it's the selected radio button.

### What is actually happening?

Focus is on Item 1.

| Environment | Info |
| --- | --- |
| antd | 5.6.3 |
| React | 18.2.0 |
| System | Windows 10 |
| Browser | Firefox 114 |

---
Tabbing within a radio group should take the focus outside the radio group, but instead, the focus moves within the radio group. Arrow keys can also not be used to select items in the radio group. The root of the issues with keyboard navigation seems to be that the input elements produced by the radio button component have no name attribute.

**Comment by afc163** [comment]
Trace #40703
